### DISCUSSION TITLE
Does a Seedvault flash drive assume 1-to-1 pairing to a device?

### DISCUSSION THREAD

**Discussion by lazytownfan** [seedvault-app/seedvault#339] [Q&A] [answered]
When using a flash drive with Seedvault, does Seedvault assume that the Seedvault backup only works with 1 device, in a 1-to-1 manner?

For example, I have a Pixel 4a with GrapheneOS with Flash Drive 1 and a Galaxy Tab S6 Lite with LineageOS with Flash Drive 2.

Is it possible to use just Flash Drive 1 for Seedvault backups with both the Pixel 4a and GT S6 Lite? Or, should I make sure to keep the flash drives separate: only use Flash Drive 1 with the Pixel 4 and Flash Drive 2 with the GT S6 Lite?

**Comment by grote** [comment, marked as answer]
As long as you don't setup the same flash drive within the same millisecond, this *should* work. But best to try it out to know for sure and maybe find hidden bugs.

**Reply by lazytownfan** [reply]
Sounds like a strong practical condition (since it realistically takes much more than 1 millisecond for Android to recognize removing the first flash drive and inserting a second one).

I'll first try this with using the same flash drive with owner and secondary profiles on the same device, and then with a different device.

**Reply by lazytownfan** [reply]
Success - I used the same flash drive for Seedvault backups with the devices mentioned above.

An interesting observation: all folder names in GrapheneOS's Seedvault backup looked hashed, while the Seedvault backup in LineageOS listed many explicit folder names for each app that was backed up. (The LOS Seedvault prompts did state that all app data was encrypted, but the contents of the Seedvault backup would reveal what the installed apps were.)

My best conjecture is that the obfuscation of installed app names must be a slight low level difference in implementation in GrapheneOS.

**Reply by grote** [reply]
The reason for the "obfuscation" is the new backup format used by Seedvault 3.

**Reply by lazytownfan** [reply]
Ok, so I'm pretty confident LineageOS won't have Seedvault 3 until LineageOS 19.0 is released... at some point.
